build.zig

```diff
@@ -1,45 +1,58 @@
 const std = @import("std");
 
-pub fn addExecutable(b: *std.Build, options: BuildStep.Options) *BuildStep {
-    return BuildStep.create(b, options);
+pub fn addExecutable(b: *std.Build, options: GoBuildStep.Options) *GoBuildStep {
+    return GoBuildStep.create(b, options);
+}
+
+/// Convenience method to create a library.
+pub fn addLibrary(b: *std.Build, options: GoBuildStep.Options) *GoBuildStep {
+    var library_options = options;
+    library_options.make_library = true;
+    // A library build requires CGO.
+    library_options.cgo_enabled = true;
+    return GoBuildStep.create(b, library_options);
 }
 
 pub fn build(b: *std.Build) void {
     _ = b;
 }
 
 /// Runs `go build` with relevant flags
-pub const BuildStep = struct {
+pub const GoBuildStep = struct {
     step: std.Build.Step,
     generated_bin: ?*std.Build.GeneratedFile,
     opts: Options,
 
     pub const Options = struct {
+        // The name of generated target.
         name: []const u8,
         target: std.Build.ResolvedTarget,
         optimize: std.builtin.OptimizeMode,
         package_path: std.Build.LazyPath,
+        // Why is this true by default? Should be false by default.
```

Comment:
I set this to true because...why use zig to build go unless you were building with CGO?